File: mobile/src/app/api/models/expense-dto.ts
/* tslint:disable */
import { ExpenseStatus } from './expense-status';
import { ServiceProvider } from './service-provider';
export interface ExpenseDto {
  adnotation?: string;
  amount: number;
  deposit: number;
  expenseStatus: ExpenseStatus;
  id: number;
  name: string;
  plannerId: number;
  provider: string;
  serviceProvider?: ServiceProvider;
}

File: WPA.backend/WPA.backend/DTOs/Budget/Expenses/ExpenseDto.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;
using WPA.backend.Entities;

namespace WPA.backend.DTOs.Expenses
{
    public class ExpenseDto
    {
        [Required]
        public int Id { get; set; }

        public ServiceProvider ServiceProvider { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public decimal Amount { get; set; }

        public string Adnotation { get; set; }

        [Required]
        public string Provider { get; set; }

        [Required]
        public decimal Deposit { get; set; }

        [Required]
        public ExpenseStatus ExpenseStatus { get; set; }

        [Required]
        public int PlannerId { get; set; }
    }
}
